explainers/beforematch-spec-draft.md

```diff
@@ -0,0 +1,51 @@
+# beforematch event
+
+_These changes will likely go into the
+[find-in-page](https://html.spec.whatwg.org/multipage/interaction.html#find-in-page)
+section of the HTML spec_
+
+When a new active match is set, either by advancing through the match list or
+due to a new find-in-page request, a `beforematch` event is fired on a node
+identified by the active match [range
+start](https://dom.spec.whatwg.org/#concept-range-start). This process follows
+the following algorithm:
+
+1. Identify the match _M_ which will become the new active match. _M_'s DOM
+range must not be [collapsed](https://dom.spec.whatwg.org/#range-collapsed).
```

Comment:
There isn't any code that looks at whether the range is collapsed. If the matching range is collapsed at the time of doing the search, then we simply wouldn't have found it, because its empty...?
Or is this not the same time as when the search is performed?
I mean I guess it makes sense to point this out, I just wonder if maybe it could be more clear that there is no need to actually check collapsed at this point? I guess saying "must not be" sort of implies this...?

Reply:
Yeah this is already naturally the case. This is an important point only because of the step that restarts at step 1 again when we're collapsed. Other implementations may consider collapsed sections as viable matches? I don't really know, but if that's the case then we'd get into an infinite loop, hence me saying that must not be collapsed at this point.

Reply:
lgtm